controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -219,7 +230,7 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
 
 	// Handle the renaming of the upstream image from "k8s.gcr.io/coredns" to "k8s.gcr.io/coredns/coredns"
 	toImageName := parsedImage.Name
-	if toImageRepository == kubernetesImageRepository && toImageName == oldCoreDNSImageName && targetMajorMinorPatch.GTE(semver.MustParse("1.8.0")) {
+	if toImageRepository == oldKubernetesImageRepository && toImageName == oldCoreDNSImageName && targetMajorMinorPatch.GTE(semver.MustParse("1.8.0")) {
```

Comment:
I wonder if we have to do this for the new registry as well, given that we could have replaced the registry above.

I think the coredns image name migration is still relevant as folks could upgrade to CoreDNS >= v1.8.0 with Kubernetes >= v1.25.0
(most folks probably never set the CoreDNS version and just get the kubeadm default, so they could recognize this relatively late and then upgrade from some very old version to a current one)

Reply:
I assumed everyone must be on coredns > v1.8.0 due to adding `EndpointSlices` which are only supported by coredns >= v1.8.1.

However for safety we can of course also do this for the new registry :-)

Reply:
But we only update the ClusterRole to add `EndpointSlices` if the CoreDNS version is actually >= v1.8.1?

I might be missing something but I think the following wouldn't have worked without handling the new registry:
* install cluster with 1.20 (CoreDNS 1.7.0)
* upgrade cluster from 1.20 until 1.25
* set CoreDNS tag to upgrade to CoreDNS >= 1.8.0

Reply:
Yeah, the other CoreDNS versions do not support EndpointSlices. I agree, which is why I added both versions to the if clause.